share/sysbuild/CMakeLists.txt

```
@@ -69,8 +69,17 @@ add_subdirectory(bootloader)
 # This allows for board and app specific images to be included.
 include(${BOARD_DIR}/sysbuild.cmake OPTIONAL)
 
-# This allows image specific sysbuild.cmake to be processed.
+# Check for any duplicate entries in image names to prevent an infinite loop
+set(TMP_IMAGES ${IMAGES})
+list(REMOVE_DUPLICATES TMP_IMAGES)
 list(LENGTH IMAGES images_length)
+list(LENGTH TMP_IMAGES tmp_images_length)
+
+if(NOT "${images_length}" EQUAL "${tmp_images_length}")
+  message(FATAL_ERROR "A duplicate image name was provided, image names must be unique")
+endif()
+
+# This allows image specific sysbuild.cmake to be processed.
 while(NOT "${images_length}" EQUAL "${processed_length}")
   foreach(image ${IMAGES})
```

It should be sufficient to do this here:
```suggestion
  list(REMOVE_DUPLICATES IMAGES)
  foreach(image ${IMAGES})
```

that will ensure that if someone adds identical image name to the list, then it will be remove before processing the list, and hence `images_length` and `processed_length` should become the same when all images has been looped.

---

Moved to inner loop and made error text more descriptive.